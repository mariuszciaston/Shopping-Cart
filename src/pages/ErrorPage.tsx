import { Link } from "react-router-dom";

const ErrorPage = () => (
  <div className="flex h-screen flex-col items-center justify-center">
    <p>This page doesn't exist.</p>
    <br />
    <Link to="/" className="font-bold">
      Go back to the home page!
    </Link>
  </div>
);

export default ErrorPage;
