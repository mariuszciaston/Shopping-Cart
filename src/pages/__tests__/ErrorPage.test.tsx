import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import ErrorPage from "../ErrorPage";
import { it, expect } from "vitest";
import { describe } from "node:test";

describe("ErrorPage", () => {
  it("renders ErrorPage with correct content", () => {
    render(
      <BrowserRouter>
        <ErrorPage />
      </BrowserRouter>,
    );

    const headingElement = screen.getByText("This page doesn't exist.");
    expect(headingElement).toBeInTheDocument();

    const linkElement = screen.getByText("Go back to the home page!");
    expect(linkElement).toBeInTheDocument();
    expect(linkElement.getAttribute("href")).toBe("/");
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <ErrorPage />
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
