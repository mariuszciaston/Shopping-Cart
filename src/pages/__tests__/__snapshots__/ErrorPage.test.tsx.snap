// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex h-screen flex-col items-center justify-center"
  >
    <p>
      This page doesn't exist.
    </p>
    <br />
    <a
      class="font-bold"
      data-discover="true"
      href="/"
    >
      Go back to the home page!
    </a>
  </div>
</DocumentFragment>
`;
